File: solutions/julia/booking-up-for-beauty/1/booking-up-for-beauty.jl
import Dates

function schedule_appointment(appointment::String)
    Dates.DateTime(appointment, "m/d/yyyy HH:MM:SS")
end

function has_passed(appointment::DateTime)
    appointment < Dates.now()
end

function is_afternoon_appointment(appointment::DateTime)
    12 <= Dates.hour(appointment) < 18
end

function describe(appointment::DateTime)
    "You have an appointment on $(Dates.format(Dates.Date(appointment), "E, U d, yyyy")) at $(Dates.format(Dates.Time(appointment), "HH:MM"))"
end

function anniversary_date()
   Dates.Date(Dates.year(Dates.now()),09,15) 
end

File: solutions/r/booking-up-for-beauty/1/booking-up-for-beauty.R
#!/usr/bin/env Rscript

suppressMessages(library(lubridate))

schedule_appointment <- function(appointment) {
  mdy_hms(appointment)
}

has_passed <- function(appointment) {
  appointment < now()
}

is_afternoon_appointment <- function(appointment) {
  appointment |> hour() >= 12 && appointment |> hour() < 18
}

day_of_week <- function(appointment) {
  wday(appointment, week_start = 1)
}

reschedule <- function(appointment) {
  appointment <- mdy_hms(appointment)
  dow <- day_of_week(appointment)
  switch(dow,
    appointment + days(4),
    appointment + days(3),
    appointment + days(2),
    appointment + days(1),
    appointment + days(7),
    appointment + days(6)
  )
}


reschedule(now() + days(1))

